bloopgun/src/main/scala/bloop/bloopgun/Bloopgun.scala

```diff
@@ -417,15 +428,22 @@ class BloopgunCli(
     def cmdWithArgs(
         found: LocatedServer,
         extraJvmOpts: List[String],
-        javaBinary: List[String] = List("java")
+        javaBinary: List[String],
+        globalSettings: GlobalSettings
     ): (List[String], Boolean) = {
       var usedExtraJvmOpts = false
       def finalJvmOpts(jvmOpts: List[String]): List[String] = {
-        if (extraJvmOpts.forall(opt => jvmOpts.contains(opt))) jvmOpts
-        else {
-          usedExtraJvmOpts = true
-          jvmOpts ++ extraJvmOpts
+        val baseOptions =
+          if (extraJvmOpts.forall(opt => jvmOpts.contains(opt))) jvmOpts
+          else {
+            usedExtraJvmOpts = true
+            jvmOpts ++ extraJvmOpts
+          }
+        val globalOptions = globalSettings.javaOptions.getOrElse(Nil)
+        if (globalOptions.nonEmpty) {
+          logger.info(s"Picked up custom Java options $globalOptions")
         }
+        baseOptions ++ globalOptions
```

Comment:
Global options have precedence over base jvm options, is that something intended?

Reply:
I added a note on why global settings take precedence.